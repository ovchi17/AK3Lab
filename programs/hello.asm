section .data:
    str hello "Hello, world!"

section .code:

start:
    psh hello

print:
    dup
    get
    jmz stop
    out
    inc
    jmp print

stop:
    drp
    drp
    hlt