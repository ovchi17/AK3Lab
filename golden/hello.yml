in_source: |-
  section .data:
    str hello "Hello, world!"

  section .code:

  start:
      psh hello

  print:
      dup
      get
      jmz stop
      out
      inc
      jmp print

  stop:
      drp
      drp
      hlt

in_stdin: |
out_log: |
  INFO     root:start.py:66 Start simulation
  DEBUG    root:start.py:26 TICK:    3 PUSH       PC:   1 DA:   0 SP:  0 TOS:   2 Stack: [2, -1, -1, -1, -1]
  DEBUG    root:start.py:26 TICK:    6 DUP        PC:   2 DA:   0 SP:  1 TOS:   2 Stack: [2, 2, -1, -1, -1]
  DEBUG    root:start.py:26 TICK:    9 LOAD       PC:   3 DA:   2 SP:  1 TOS:  72 Stack: [2, 72, -1, -1, -1]
  DEBUG    root:start.py:26 TICK:   11 JZ         PC:   4 DA:   2 SP:  1 TOS:  72 Stack: [2, 72, -1, -1, -1]
  DEBUG    root:start.py:26 TICK:   15 PUSH2      PC:   5 DA:   2 SP:  2 TOS:  72 Stack: [2, 1, 72, -1, -1]
  DEBUG    root:datapath.py:125 output: H << 72
  DEBUG    root:start.py:26 TICK:   18 POP        PC:   6 DA:   1 SP:  0 TOS:   2 Stack: [2, 1, 72, -1, -1]
  DEBUG    root:start.py:26 TICK:   20 INC        PC:   7 DA:   1 SP:  0 TOS:   3 Stack: [3, 1, 72, -1, -1]
  DEBUG    root:start.py:26 TICK:   22 JUMP       PC:   1 DA:   1 SP:  0 TOS:   3 Stack: [3, 1, 72, -1, -1]
  DEBUG    root:start.py:26 TICK:   25 DUP        PC:   2 DA:   1 SP:  1 TOS:   3 Stack: [3, 3, 72, -1, -1]
  DEBUG    root:start.py:26 TICK:   28 LOAD       PC:   3 DA:   3 SP:  1 TOS:  101 Stack: [3, 101, 72, -1, -1]
  DEBUG    root:start.py:26 TICK:   30 JZ         PC:   4 DA:   3 SP:  1 TOS:  101 Stack: [3, 101, 72, -1, -1]
  DEBUG    root:start.py:26 TICK:   34 PUSH2      PC:   5 DA:   3 SP:  2 TOS:  101 Stack: [3, 1, 101, -1, -1]
  DEBUG    root:datapath.py:125 output: He << 101
  DEBUG    root:start.py:26 TICK:   37 POP        PC:   6 DA:   1 SP:  0 TOS:   3 Stack: [3, 1, 101, -1, -1]
  DEBUG    root:start.py:26 TICK:   39 INC        PC:   7 DA:   1 SP:  0 TOS:   4 Stack: [4, 1, 101, -1, -1]
  DEBUG    root:start.py:26 TICK:   41 JUMP       PC:   1 DA:   1 SP:  0 TOS:   4 Stack: [4, 1, 101, -1, -1]
  DEBUG    root:start.py:26 TICK:   44 DUP        PC:   2 DA:   1 SP:  1 TOS:   4 Stack: [4, 4, 101, -1, -1]
  DEBUG    root:start.py:26 TICK:   47 LOAD       PC:   3 DA:   4 SP:  1 TOS:  108 Stack: [4, 108, 101, -1, -1]
  DEBUG    root:start.py:26 TICK:   49 JZ         PC:   4 DA:   4 SP:  1 TOS:  108 Stack: [4, 108, 101, -1, -1]
  DEBUG    root:start.py:26 TICK:   53 PUSH2      PC:   5 DA:   4 SP:  2 TOS:  108 Stack: [4, 1, 108, -1, -1]
  DEBUG    root:datapath.py:125 output: Hel << 108
  DEBUG    root:start.py:26 TICK:   56 POP        PC:   6 DA:   1 SP:  0 TOS:   4 Stack: [4, 1, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   58 INC        PC:   7 DA:   1 SP:  0 TOS:   5 Stack: [5, 1, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   60 JUMP       PC:   1 DA:   1 SP:  0 TOS:   5 Stack: [5, 1, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   63 DUP        PC:   2 DA:   1 SP:  1 TOS:   5 Stack: [5, 5, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   66 LOAD       PC:   3 DA:   5 SP:  1 TOS:  108 Stack: [5, 108, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   68 JZ         PC:   4 DA:   5 SP:  1 TOS:  108 Stack: [5, 108, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   72 PUSH2      PC:   5 DA:   5 SP:  2 TOS:  108 Stack: [5, 1, 108, -1, -1]
  DEBUG    root:datapath.py:125 output: Hell << 108
  DEBUG    root:start.py:26 TICK:   75 POP        PC:   6 DA:   1 SP:  0 TOS:   5 Stack: [5, 1, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   77 INC        PC:   7 DA:   1 SP:  0 TOS:   6 Stack: [6, 1, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   79 JUMP       PC:   1 DA:   1 SP:  0 TOS:   6 Stack: [6, 1, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   82 DUP        PC:   2 DA:   1 SP:  1 TOS:   6 Stack: [6, 6, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   85 LOAD       PC:   3 DA:   6 SP:  1 TOS:  111 Stack: [6, 111, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   87 JZ         PC:   4 DA:   6 SP:  1 TOS:  111 Stack: [6, 111, 108, -1, -1]
  DEBUG    root:start.py:26 TICK:   91 PUSH2      PC:   5 DA:   6 SP:  2 TOS:  111 Stack: [6, 1, 111, -1, -1]
  DEBUG    root:datapath.py:125 output: Hello << 111
  DEBUG    root:start.py:26 TICK:   94 POP        PC:   6 DA:   1 SP:  0 TOS:   6 Stack: [6, 1, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:   96 INC        PC:   7 DA:   1 SP:  0 TOS:   7 Stack: [7, 1, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:   98 JUMP       PC:   1 DA:   1 SP:  0 TOS:   7 Stack: [7, 1, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  101 DUP        PC:   2 DA:   1 SP:  1 TOS:   7 Stack: [7, 7, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  104 LOAD       PC:   3 DA:   7 SP:  1 TOS:  44 Stack: [7, 44, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  106 JZ         PC:   4 DA:   7 SP:  1 TOS:  44 Stack: [7, 44, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  110 PUSH2      PC:   5 DA:   7 SP:  2 TOS:  44 Stack: [7, 1, 44, -1, -1]
  DEBUG    root:datapath.py:125 output: Hello, << 44
  DEBUG    root:start.py:26 TICK:  113 POP        PC:   6 DA:   1 SP:  0 TOS:   7 Stack: [7, 1, 44, -1, -1]
  DEBUG    root:start.py:26 TICK:  115 INC        PC:   7 DA:   1 SP:  0 TOS:   8 Stack: [8, 1, 44, -1, -1]
  DEBUG    root:start.py:26 TICK:  117 JUMP       PC:   1 DA:   1 SP:  0 TOS:   8 Stack: [8, 1, 44, -1, -1]
  DEBUG    root:start.py:26 TICK:  120 DUP        PC:   2 DA:   1 SP:  1 TOS:   8 Stack: [8, 8, 44, -1, -1]
  DEBUG    root:start.py:26 TICK:  123 LOAD       PC:   3 DA:   8 SP:  1 TOS:  32 Stack: [8, 32, 44, -1, -1]
  DEBUG    root:start.py:26 TICK:  125 JZ         PC:   4 DA:   8 SP:  1 TOS:  32 Stack: [8, 32, 44, -1, -1]
  DEBUG    root:start.py:26 TICK:  129 PUSH2      PC:   5 DA:   8 SP:  2 TOS:  32 Stack: [8, 1, 32, -1, -1]
  DEBUG    root:datapath.py:125 output: Hello,  << 32
  DEBUG    root:start.py:26 TICK:  132 POP        PC:   6 DA:   1 SP:  0 TOS:   8 Stack: [8, 1, 32, -1, -1]
  DEBUG    root:start.py:26 TICK:  134 INC        PC:   7 DA:   1 SP:  0 TOS:   9 Stack: [9, 1, 32, -1, -1]
  DEBUG    root:start.py:26 TICK:  136 JUMP       PC:   1 DA:   1 SP:  0 TOS:   9 Stack: [9, 1, 32, -1, -1]
  DEBUG    root:start.py:26 TICK:  139 DUP        PC:   2 DA:   1 SP:  1 TOS:   9 Stack: [9, 9, 32, -1, -1]
  DEBUG    root:start.py:26 TICK:  142 LOAD       PC:   3 DA:   9 SP:  1 TOS:  119 Stack: [9, 119, 32, -1, -1]
  DEBUG    root:start.py:26 TICK:  144 JZ         PC:   4 DA:   9 SP:  1 TOS:  119 Stack: [9, 119, 32, -1, -1]
  DEBUG    root:start.py:26 TICK:  148 PUSH2      PC:   5 DA:   9 SP:  2 TOS:  119 Stack: [9, 1, 119, -1, -1]
  DEBUG    root:datapath.py:125 output: Hello, w << 119
  DEBUG    root:start.py:26 TICK:  151 POP        PC:   6 DA:   1 SP:  0 TOS:   9 Stack: [9, 1, 119, -1, -1]
  DEBUG    root:start.py:26 TICK:  153 INC        PC:   7 DA:   1 SP:  0 TOS:  10 Stack: [10, 1, 119, -1, -1]
  DEBUG    root:start.py:26 TICK:  155 JUMP       PC:   1 DA:   1 SP:  0 TOS:  10 Stack: [10, 1, 119, -1, -1]
  DEBUG    root:start.py:26 TICK:  158 DUP        PC:   2 DA:   1 SP:  1 TOS:  10 Stack: [10, 10, 119, -1, -1]
  DEBUG    root:start.py:26 TICK:  161 LOAD       PC:   3 DA:  10 SP:  1 TOS:  111 Stack: [10, 111, 119, -1, -1]
  DEBUG    root:start.py:26 TICK:  163 JZ         PC:   4 DA:  10 SP:  1 TOS:  111 Stack: [10, 111, 119, -1, -1]
  DEBUG    root:start.py:26 TICK:  167 PUSH2      PC:   5 DA:  10 SP:  2 TOS:  111 Stack: [10, 1, 111, -1, -1]
  DEBUG    root:datapath.py:125 output: Hello, wo << 111
  DEBUG    root:start.py:26 TICK:  170 POP        PC:   6 DA:   1 SP:  0 TOS:  10 Stack: [10, 1, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  172 INC        PC:   7 DA:   1 SP:  0 TOS:  11 Stack: [11, 1, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  174 JUMP       PC:   1 DA:   1 SP:  0 TOS:  11 Stack: [11, 1, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  177 DUP        PC:   2 DA:   1 SP:  1 TOS:  11 Stack: [11, 11, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  180 LOAD       PC:   3 DA:  11 SP:  1 TOS:  114 Stack: [11, 114, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  182 JZ         PC:   4 DA:  11 SP:  1 TOS:  114 Stack: [11, 114, 111, -1, -1]
  DEBUG    root:start.py:26 TICK:  186 PUSH2      PC:   5 DA:  11 SP:  2 TOS:  114 Stack: [11, 1, 114, -1, -1]
  DEBUG    root:datapath.py:125 output: Hello, wor << 114
  DEBUG    root:start.py:26 TICK:  189 POP        PC:   6 DA:   1 SP:  0 TOS:  11 Stack: [11, 1, 114, -1, -1]
  DEBUG    root:start.py:26 TICK:  191 INC        PC:   7 DA:   1 SP:  0 TOS:  12 Stack: [12, 1, 114, -1, -1]
  DEBUG    root:start.py:26 TICK:  193 JUMP       PC:   1 DA:   1 SP:  0 TOS:  12 Stack: [12, 1, 114, -1, -1]
  DEBUG    root:start.py:26 TICK:  196 DUP        PC:   2 DA:   1 SP:  1 TOS:  12 Stack: [12, 12, 114, -1, -1]
  DEBUG    root:start.py:26 TICK:  199 LOAD       PC:   3 DA:  12 SP:  1 TOS:  108 Stack: [12, 108, 114, -1, -1]
  DEBUG    root:datapath.py:125 output: Hello, worl << 108
  DEBUG    root:datapath.py:125 output: Hello, world << 100
  DEBUG    root:datapath.py:125 output: Hello, world! << 33
  WARNING  root:start.py:33 Halt ere my magnificence, mortal, lest thou be consumed by the brilliance of my splendor
  INFO     root:start.py:43 output_buffer: Hello, world!
  INFO     root:start.py:75 End simulation
out_stdout: |
  LoC: 20 Code instr: 28
  ============================================================
  Hello, world!
  Instructions: 98 Ticks: 263
out_code: |-
  [16,
   0,
   0,
   72,
   101,
   108,
   108,
   111,
   44,
   32,
   119,
   111,
   114,
   108,
   100,
   33,
   0,
   {"index": 0, "opcode": "psh", "arg": 2},
   {"index": 1, "opcode": "dup", "arg": null},
   {"index": 2, "opcode": "get", "arg": null},
   {"index": 3, "opcode": "jmz", "arg": 8},
   {"index": 4, "opcode": "psh2", "arg": 1},
   {"index": 5, "opcode": "pop", "arg": null},
   {"index": 6, "opcode": "inc", "arg": null},
   {"index": 7, "opcode": "jmp", "arg": 1},
   {"index": 8, "opcode": "drp", "arg": null},
   {"index": 9, "opcode": "drp", "arg": null},
   {"index": 10, "opcode": "hlt", "arg": null}]
